src/pruna/algorithms/base/pruna_base.py

```diff
@@ -278,6 +268,59 @@ def apply(self, model: Any, smash_config: SmashConfig) -> Any:
         wrapped_config = SmashConfigPrefixWrapper(smash_config, prefix)
         return self._apply(model, wrapped_config)
 
+    def _expand_tags_into_algorithm_names(self, items: Iterable[str | AlgorithmTag] | None) -> list[str]:
+        """Expand algorithms/tags -> concrete algorithm names."""
+        if not items:
+            return []
+
+        # avoid circular import
+        from pruna.algorithms import AlgorithmRegistry
+
+        out: list[str] = []
+        for it in items:
+            if isinstance(it, str):
+                out.append(it)
+            else:
+                out.extend(AlgorithmRegistry.get_algorithms_by_tag(it))
+        return out
+
+    def get_incompatible_algorithms(self) -> list[str]:
+        """
+        Get algorithms incompatible with this one.
+
+        Returns
+        -------
+        list[str]
+            The incompatible algorithms.
+        """
+        cb = set(self._expand_tags_into_algorithm_names(self.compatible_before))
+        ca = set(self._expand_tags_into_algorithm_names(self.compatible_after))
+        allowed = cb | ca
```

Comment:
So nitpicky i'm doing it myself - but yes clearer variable naming :melting_face:
```suggestion
        compatible_before = set(self._expand_tags_into_algorithm_names(self.compatible_before))
        compatible_after = set(self._expand_tags_into_algorithm_names(self.compatible_after))
        allowed = compatible_before | compatible_after
```